src/System.Private.CoreLib/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -191,6 +191,56 @@ internal static SignatureHelper GetTypeSigToken(Module module, Type type)
 
             return new SignatureHelper(module, type);
         }
+
+        internal static void ExtractParametersTypeArrays(ParameterInfo[] parameters,
+                                                         out Type[] parameterTypes,
+                                                         out Type[][] parameterTypeRequiredCustomModifiers,
+                                                         out Type[][] parameterTypeOptionalCustomModifiers)
+        {
+            // This method tries to allocate as few arrays as possible. In the absence of parameters or
+            // custom modifiers, arrays or single items will default to null. The returned arrays will
+            // generally be fed back to a `SignatureHelper` which must perform null checks where necessary.
+
+            Type[] types = null;
+            Type[][] requiredCustomModifiers = null;
+            Type[][] optionalCustomModifiers = null;
+
+            if (parameters?.Length > 0)
```

Comment:
Every caller of this passes a non-null `parameters`. The `?` isn't needed.

Reply:
Thanks for the review. I've replaced all three null-conditional member accesses `?.` with `.`.